pytorchfi/core.py

```diff
@@ -44,7 +45,7 @@ def __init__(self, model, batch_size, input_shape=None, layer_types=None, **kwar
         if len(layer_types) < 0:
             raise AssertionError("Error: At least one layer type must be selected.")
 
-        handles, _shapes = self._traverse_model_set_hooks(
+        handles, _shapes, self.weights_size = self._traverse_model_set_hooks(
```

Comment:
Currently `_shapes` isn't being used – is this something we need to return from `self._traverse_model_set_hooks`?

Reply:
It's used internally in the recursive call. Let's address with next PR to clean this up as suggested in next comment.